adapters/mobfoxpb/mobfoxpb.go

```diff
@@ -28,21 +39,40 @@ func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters
 // MakeRequests create bid request for mobfoxpb demand
 func (a *adapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
 	var errs []error
-	var err error
 	var tagID string
-
+	var err_tag error
+	var key string
+	var err_key error
+	var route string
+	var method string
 	var adapterRequests []*adapters.RequestData
 
+	requestURI := a.URI
 	reqCopy := *request
 	imp := request.Imp[0]
-	tagID, err = jsonparser.GetString(imp.Ext, "bidder", "TagID")
-	if err != nil {
-		errs = append(errs, err)
+	tagID, err_tag = jsonparser.GetString(imp.Ext, "bidder", "TagID")
+	key, err_key = jsonparser.GetString(imp.Ext, "bidder", "key")
+	if err_tag != nil && err_key != nil {
+		errs = append(errs, &errortypes.BadInput{
+			Message: fmt.Sprintf("Invalid or non existing key or tagId, atleast one should be present"),
```

Comment:
From the code it seems like you expect either a valid key or tag ID and so you might want the error message to be:
```
Invalid or non existing key and tagId
```

Reply:
Not really sure what do you mean, to remove this part from message `, atleast one should be present` ?

Reply:
Nope! Sorry that wasn't super clear. I meant changing the `key or tagID` part to `key and tagID` because you will get to this error only if both `key` and `tagId` are invalid. Hope that makes sense

Reply:
We really expect atleast one parameter, and fail only if both not present or invalid

Reply:
Yeah which is why that should be `Invalid or non existing key and tagID` because that'd indicate that both are either not present or invalid and therefore it's failing. `Invalid or non existing key or tagID` indicates that you need both those values to be present and valid.